opta/core/gcp.py

```diff
@@ -173,3 +167,32 @@ def bucket_exists(self, bucket_name: str) -> bool:
         except NotFound:
             return False
         return True
+
+    @classmethod
+    def get_all_remote_configs(cls) -> Dict[str, Dict[str, "StructuredConfig"]]:
+        prefix = "opta_config/"
+        credentials, project_id = cls.get_credentials()
+        storage_client = storage.Client(project=project_id, credentials=credentials)
+        opta_config_detailed_map = {}
+        for bucket in storage_client.list_buckets(prefix="opta-tf-state"):
+            detailed_config = {}
+            for response in storage_client.list_blobs(
+                bucket.name, prefix=prefix, delimiter="/"
+            ):
+                structured_config = cls._download_remote_config(bucket, response.name)
+                if structured_config:
+                    detailed_config[response.name[len(prefix) :]] = structured_config
+            if detailed_config:
+                opta_config_detailed_map[bucket.name] = detailed_config
+        return opta_config_detailed_map
+
+    @staticmethod
+    def _download_remote_config(bucket: Bucket, key: str) -> Optional["StructuredConfig"]:
```

Comment:
key is a `path`

Reply:
In bucket terminology, `key` is better understood as compared to the path.

Reply:
that's fine, it looks like aws uses key with gcp uses name.
gcp storage.Blob() has a param `encryption_key` so that's why I was suggesting a different name. but I didn't realize that AWS was using key.